cmd/clair/main.go

```diff
@@ -35,32 +35,51 @@ const (
 
 func main() {
 	// parse conf from cli
-	var (
-		confFile ConfValue
-		conf     config.Config
-		runMode  ConfMode
-	)
-	confFile.Set(os.Getenv(envConfig))
-	runMode.Set(os.Getenv(envMode))
-	flag.Var(&confFile, "conf", "The file system path to Clair's config file.")
-	flag.Var(&runMode, "mode", "The operation mode for this server.")
+	var conf config.Config
+	flag.String("conf", "", "The file system path to Clair's config file.")
+	flag.String("mode", "combo", "The operation mode for this server.")
 	flag.Parse()
-	if confFile.String() == "" {
-		golog.Fatalf("must provide a -conf flag or set %q in the environment", envConfig)
-	}
+	flag.VisitAll(func(f *flag.Flag) {
+		fv := f.Value.(flag.Getter).Get().(string)
+		var key string
+		switch f.Name {
+		case "conf":
+			key = envConfig
+		case "mode":
+			key = envMode
+		}
+		v, ok := os.LookupEnv(key)
+		if fv == "" && !ok {
+			golog.Fatalf("must provide a -%s value or set %q in the environment", f.Name, key)
+		}
+		if fv == "" && ok {
+			fv = v
+		}
+		switch f.Name {
+		case "conf":
+			cf, err := os.Open(fv)
+			if err != nil {
+				golog.Fatalf("failed to open config file: %v", err)
+			}
+			defer cf.Close()
```

Comment:
Maybe w/o the defer and a few lines down, but not sure it matters

Reply:
I don't think it matters